app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt

```diff
@@ -368,11 +374,68 @@ class StateFragmentTest {
             "Incidentally, do you know where the name 'Oppia' comes from?"
-      onView(atPositionOnView(R.id.state_recycler_view, 0,R.id.content_text_view)).check(matches(hasDescendant(withText(htmlResult))))
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(matches(withText(htmlResult)))
+    }
+  }
+
+  @Test
+  fun testMultipleChoiceInput_showsRadioButtons_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOption() {
+    launch(ContentCardTestActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button_1)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.MultipleChoiceViewHolder>(0, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.MultipleChoiceViewHolder>(1, click())
+      )
+    }
+  }
+
+  @Test
+  fun testItemSelectionInput_showsCheckBox_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOptions() {
+    launch(ContentCardTestActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button_2)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(0, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(2, click())
+      )
+    }
+  }
+
+  @Test
+  fun testItemSelectionInput_showsCheckBox_withMaxSelectionAllowed_userSelectsDesiredOptions_correctError() {
+    launch(ContentCardTestActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button_2)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(0, click())
+      )
+      it.onActivity { activity ->
+
+        activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+        activity.requestedOrientation = Configuration.ORIENTATION_PORTRAIT
+      }
+      counter++
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(2, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(1, click())
+      )
+      counter++
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(3, click())
+      )
+      counter++
+      TestCase.assertTrue(
+        "Error, You cannot select more than $maxSelectionAllowedCount",
+        counter >= maxSelectionAllowedCount
```

Comment:
Does this actually check the number of selections? It seems like the above could be radio buttons and this would still be counting for each interaction that's successful. I suggest removing this, and if we care to check the number of selections we should probably rely on the submitted answer.

Reply:
Removed this assert check.